iron-input.html

```diff
@@ -118,15 +118,25 @@
     },
 
     ready: function() {
-      this.bindValue = this.value;
+      if (this.type === 'number') {
+        this.bindValue = parseFloat(this.value);
```

Comment:
Could be simplified to `this.bindValue = this.type === 'number' ? parseFloat(this.value) : this.value;`